### PR TITLE
[1pt] PR: Update instances of pandas.append (deprecated) to pandas.concat

### LINKED ISSUE #898: [5pt] Update scripts that contain append statements for dataframes
After a recent docker image update, some scripts need to be updated. `tools_shared_functions.py` is one of these. There may be more.

### PR DESCRIPTION
**Pull request by mluck**
`pandas.append` was deprecated in our last Pandas upgrade (#889). This PR updates the remaining instances of `pandas.append` to `pandas.concat`. Closes #898.

The file `tools/thalweg_drop_check.py` had an instance of `pandas.append` but was deleted as it is no longer used or necessary.

### Changes

The following files had instances of `pandas.append` changed to `pandas.concat`:
- `data/`
    - `nws/preprocess_ahps_nws.py`
    - `usgs/`
        - `acquire_and_preprocess_3dep_dems.py`
        - `preprocess_ahps_usgs.py`
- `src/`
    - `add_crosswalk.py`
    - `adjust_headwater_streams.py`
    - `aggregate_vector_inputs.py`
    - `reset_mannings.py`
- `tools/`
    - `aggregate_mannings_calibration.py`
    - `eval_plots.py`
    - `generate_categorical_fim.py`
    - `generate_categorical_fim_flows.py`
    - `plots/`
        - `eval_plots.py`
        - `utils/shared_functions.py`
    - `rating_curve_comparison.py`
    - `rating_curve_get_usgs_curves.py`
    - `tools_shared_functions.py`

### Removals

- `tools/thalweg_drop_check.py`

### Testing

Ran `fim_pipeline.sh`, `synthesize_test_cases.py`, and `rating_curve_comparison.py` on 12040101

### Checklist

_You may update this checklist before and/or after creating the PR. If you're unsure about any of them, please ask, we're here to help! These items are what we are going to look for before merging your code._

- [x] Informative and human-readable title, using the format: `[_pt] PR: <description>`
- [x] Links are provided if this PR resolves an issue, or depends on another other PR
- [x] If submitting a PR to the `dev` branch (the default branch), you have a descriptive [Feature Branch](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow) name using the format: `dev-<description-of-change>` (e.g.: `dev-revise-levee-masking`)
- [x] Changes are limited to a single goal (no scope creep)
- [x] The feature branch you're submitting as a PR is up to date (merged) with the latest `dev` branch
- [x] Changes adhere to [PEP-8 Style Guidelines](https://peps.python.org/pep-0008/)
- [x] Any _change_ in functionality is tested
- [x] Passes all unit tests locally (inside interactive Docker container, at `/foss_fim/`, run: `pytest unit_tests/`)
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [ ] Project documentation has been updated ([CHANGELOG](/docs/CHANGELOG.md) and/or [README](/README.md))
- [x] [Reviewers requested](https://help.github.com/articles/requesting-a-pull-request-review/)

### CODE REVIEW

**Comment by BradfordBates-NOAA** [comment]
Nice work. I'll run a few more tests but overall this looks great.

**Review by BradfordBates-NOAA — DISMISSED** [review]

**Review by BradfordBates-NOAA — APPROVED** [review]

**Comment by RobHanna-NOAA** [comment]
@BradfordBates-NOAA , I encourage you not to merge during the week but wait until the weekly Friday merge, unless the current merge is an critical emergency.  I need to be able to keep track of merges and versioning as well as have a rough understanding of what is in each build so I know what to expect during the run.  People can merge other branches into their current work in progress branch if  that is needed, but also not always encourages as the other branch may not yet be tested.

**Comment by RobHanna-NOAA** [comment]
How is that for ironic. We did need it now to get the sierra test to run. Well.. you get the point. :).

**Pull request merged**
